app/scripts/models/reliers/oauth.js

```diff
@@ -204,6 +207,41 @@ define(function (require, exports, module) {
     },
 
     /**
+     * Check if the relier wants access to the account encryption keys.
+     *
+     * @returns {Boolean}
+     */
+    wantsKeys () {
+      return !! (this.has('keysJwk') && this._config.scopedKeysEnabled && this._validateKeyScope());
+    },
+
+    /**
+     * Validate the requested scope with the relier redirect uri
+     * @returns {boolean}
+     * @private
+     */
+    _validateKeyScope () {
+      const validation = this._config.scopedKeysValidation || {};
+
+      if (! this.get('scope')) {
+        throw new Error('Invalid scope parameter');
+      }
+
+      const scopes = scopeStrToArray(this.get('scope'));
+
+      scopes.forEach((scope) => {
+        const existingScope = validation.hasOwnProperty(scope);
+
+        if (existingScope && ! validation[scope].redirectUris.includes(this.get('redirectUri'))) {
+          throw new Error('Invalid redirect parameter');
+        }
+      });
+
+      return true;
+    },
```

Comment:
I like this, but I think we should add one more rule: you have to match *at least one scope* in the validation set for it to return true.

So for example, if the relier specifies `keys_jwk=BLAHBLAH&scope=profile`, then we should fail out rather than returning `true` here.  The relier is not requesting any key-bearing scopes so we shouldn't fetch the keys at all.

Reply:
Good idea 👍